Feature/450 improve deployment log

mburri opened the pull request:
- [x] new rest endpoint to fetch available log files from backend
- [x] implement deployment log view with angular
- [x] delete legacy view (jsf) and bean
- [x] highlight some keywords with codemirror - more keywords possible
- [x] make file selection a little bit more obvious

fixes #450 and also fixes #443

mburri commented:
The lates commit highlights whole lines where certain keywords appear:
![image](https://user-images.githubusercontent.com/4577216/103762465-a0137900-5018-11eb-84d7-2e2fba6901ae.png)

yvespp commented:
@mburri thanks for the work so far. Here my feedback:
* Can you please add some space between the log file selector and the log box?
* In my chrome the logfile box doesn't use the whole screen vertically. Can you make it so it uses the whole screen height? ![image](https://user-images.githubusercontent.com/15231595/103879029-969d1600-50d7-11eb-87a0-69680207b637.png)
  * Maybe the gray box around the log could be removed and the scrollbar of the browser could be used instead?
* Maybe the title `Deployments` could be removed and replace with `Log file for 709104`
* Can you turn of the line numbering and enable the search like in the old log view?

mburri commented:
- [x] added some space between file selector and codemirror box
- [x] use full screen height
- [x] changed page title
- [x] turn of line numbering
- [x] enable codemirror search box
- [x] fixed markup to remove grey box around log and use the browsers sroll bar
  
![image](https://user-images.githubusercontent.com/4577216/103982941-b6d6de80-5184-11eb-9777-ab60b35ff000.png)

yvespp commented:
Can make it look like this please?
![image](https://user-images.githubusercontent.com/15231595/104018574-4945a500-51ba-11eb-88a0-1df59ba27ed0.png)

yvespp commented:
Somehow the log box still doesn't use the whole height of the screen:
![image](https://user-images.githubusercontent.com/15231595/104165357-89df3180-53f9-11eb-9818-95370bdd2881.png)

Tried in chrome and IE.

yvespp commented on AMW_angular/io/src/app/deployments/deployments-list.component.html:
```diff
@@ -75,7 +75,10 @@
     </thead>
     <tbody>
-      <tr *ngFor="let deployment of deployments" class="deploy-{{ deployment.state }}">
+      <tr
```
For me this formatting is harder to read... does you IDE automatically do this?

mburri replied:
I'm using prettier as code formatter, yes. 
what's the problem exactly:
- that attributes are placed on a new line?
- the closing bracket that is on a new line?

yvespp replied:
That the line is split into multiple lines even though it would fit on one line.

yvespp reviewed (changes requested):
For the rest service:
I think it would be better if `/deployments/{id}/logs` would only return the info which logs exist and what their names are (only metadata). The logs can then be received with `/deployments/{id}/logs/{logid/logName}`.
Why:
* The log file doesn't have to be escaped to fit into json.
* `/deployments/{id}/logs/{logid}` can return just plain text and a client doesn't have to be able to read json to read the log.
* The log view must only load the file that is visible.

mburri commented:
as requested: 

- [x] raise max. number of characters per line (prettier)
- [x] changed backend api (get log file content as text/plain
- [x] highlight warning messages
also improved:
- [x] set log file name the url to allow direct links

yvespp commented:
Thanks for you work and patience!
